### PR TITLE
Azure: Build peerpod vm image using packer with limited scope

### PR DESCRIPTION
**Pull request by kartikjoshi21**
Update packer config to build pod vm image using resource group specific scope

### CODE REVIEW

**Review by knrt10 — APPROVED** [review]
+1

File: azure/README.md [hunk 1]
```diff
@@ -6,7 +6,6 @@
 
 ```bash
 export RESOURCE_GROUP="REPLACE_ME"
-export LOCATION="REPLACE_ME"
```
Comment (surajssd) [inline azure/README.md:9]: The subsequent command will fail to create resource group without providing location?

File: azure/README.md [hunk 2]
```diff
@@ -16,15 +15,20 @@ az group create --name "${RESOURCE_GROUP}" --location "${LOCATION}"
...
-  --query "{ client_id: appId, client_secret: password, tenant_id: tenant }"
+az ad sp create-for-rbac
+  --name "Packer Build"
+  --role "Contributor"
+  --scopes /subscriptions/${SUBSCRIPTION_ID}/resourceGroups/${RESOURCE_GROUP}
+
+az role assignment create
+  --assignee ${CLIENT_ID}
+  --role "Contributor"
+  --scope /subscriptions/${SUBSCRIPTION_ID}/resourceGroups/${RESOURCE_GROUP}
```
Comment (surajssd) [inline azure/README.md:26]:
```suggestion
az ad sp create-for-rbac \
    --name "Packer Build" \
    --role "Contributor" \
    --scopes /subscriptions/${SUBSCRIPTION_ID}/resourceGroups/${RESOURCE_GROUP} \
    --query "{ CLIENT_ID: appId, CLIENT_SECRET: password, TENANT_ID: tenant }"

az role assignment create \
  --assignee ${CLIENT_ID} \
  --role "Contributor" \
  --scope /subscriptions/${SUBSCRIPTION_ID}/resourceGroups/${RESOURCE_GROUP}
```

Comment (surajssd) [reply azure/README.md:26]: Also the last command fails with the following error:

```bash
$ az role assignment create \
  --assignee ${CLIENT_ID} \
  --role "Contributor" \
  --scope /subscriptions/${SUBSCRIPTION_ID}/resourceGroups/${RESOURCE_GROUP}
argument --assignee: expected one argument
```

My guess is that `CLIENT_ID` is not populated, the new doc should mention where this `CLIENT_ID` is coming from.

Comment (surajssd) [reply azure/README.md:26]: I think I will divide the above two commands with instructions between them to export respective environment variable.

**Review by surajssd — CHANGES_REQUESTED** [review]

**Review by surajssd — CHANGES_REQUESTED** [review]

File: azure/README.md [hunk 3]
```diff
@@ -16,22 +16,32 @@ az group create --name "${RESOURCE_GROUP}" --location "${LOCATION}"
...
+  --name "Packer Build"
+  --role "Contributor"
+  --scopes /subscriptions/${SUBSCRIPTION_ID}/resourceGroups/${RESOURCE_GROUP}
 ```
 
+
 - Set environment variables
 
-The env var `CLIENT_ID`, `CLIENT_SECRET`, `TENANT_ID` can be copied from the output of the last command:
+The env var `CLIENT_ID`:`appId`, `CLIENT_SECRET`:`password`, `TENANT_ID`:`tenant` can be copied from the output of the last command:
```
[the remark this answers is not in the document]
Comment (kartikjoshi21) [reply azure/README.md:28]: This command actually gives all the things in appid, password and tenant format

[the remark this answers is not in the document]
Comment (surajssd) [reply azure/README.md:28]: That's why I suggested this change:

```bash
az ad sp create-for-rbac \
    --name "Packer Build" \
    --role "Contributor" \
    --scopes /subscriptions/${SUBSCRIPTION_ID}/resourceGroups/${RESOURCE_GROUP} \
    --query "{ CLIENT_ID: appId, CLIENT_SECRET: password, TENANT_ID: tenant }"
```

That way the user does not have to do the manual translation.

**Review by surajssd — APPROVED** [review]

**Pull request merged**